src/main/kotlin/org/hyacinthbots/lilybot/utils/_Utils.kt

```diff
@@ -269,6 +264,136 @@ suspend inline fun CheckContext<*>.configPresent(vararg configOptions: ConfigOpt
+}
+
+/**
+ * This function checks if a single config exists and is valid. Returns true if it is or false otherwise.
+ *
+ * @param channelType The type of logging channel desired
+ * @param guild The guild the desired channel is in
+ * @return The logging channel of [channelType] for the [guild] or null if it doesn't exist
+ * @author tempest15
+ * @since 4.1.0
+ */
+suspend inline fun getLoggingChannelWithPerms(channelType: ConfigOptions, guild: GuildBehavior): GuildMessageChannel? {
+	val guildId = guild.id
+
+	if (!configIsUsable(channelType, guildId)) return null
+
+	val channelId = when (channelType) {
+		ConfigOptions.SUPPORT_CHANNEL -> SupportConfigCollection().getConfig(guildId)?.channel ?: return null
+		ConfigOptions.ACTION_LOG -> ModerationConfigCollection().getConfig(guildId)?.channel ?: return null
+		ConfigOptions.UTILITY_LOG -> UtilityConfigCollection().getConfig(guildId)?.utilityLogChannel ?: return null
+		ConfigOptions.MESSAGE_LOG -> LoggingConfigCollection().getConfig(guildId)?.messageChannel ?: return null
+		ConfigOptions.MEMBER_LOG -> LoggingConfigCollection().getConfig(guildId)?.memberLog ?: return null
+		else -> throw IllegalArgumentException("$channelType does not point to a channel.")
+	}
+	val channel = guild.getChannelOfOrNull<GuildMessageChannel>(channelId) ?: return null
+
+	if (!channel.botHasPermissions(Permission.ViewChannel) || !channel.botHasPermissions(Permission.SendMessages)) {
+		when (channelType) {
+			ConfigOptions.SUPPORT_CHANNEL -> SupportConfigCollection().clearConfig(guildId)
+			ConfigOptions.ACTION_LOG -> ModerationConfigCollection().clearConfig(guildId)
+			ConfigOptions.UTILITY_LOG -> UtilityConfigCollection().clearConfig(guildId)
+			ConfigOptions.MESSAGE_LOG -> LoggingConfigCollection().clearConfig(guildId)
+			ConfigOptions.MEMBER_LOG -> LoggingConfigCollection().clearConfig(guildId)
+			else -> throw IllegalArgumentException("$channelType does not point to a channel.")
+		}
+		getFirstUsableChannel(guild)?.createMessage(
+			"Lily is unable to send messages in the configured " +
+					"${channelType.toString().lowercase()} for this guild. " +
+					"As a result, the corresponding config has been reset. \n\n" +
+					"*Note:* this channel has been used to send this message because it's the first channel " +
+					"in the guild Lily could use. Please inform this guild's staff about this message."
+		)
+		return null
+	}
+
+	return channel
+}
+
+/**
+ * Get the first text channel the bot can send a message in.
+ *
+ * @param inputGuild The guild in which to get the channel.
+ * @return The first text channel the bot can send a message in or null if there isn't one.
+ * @author tempest15
+ * @since 3.5.4
+ */
+suspend inline fun getFirstUsableChannel(inputGuild: GuildBehavior): GuildMessageChannel? {
+	val channelWithPerms = inputGuild.channels.first {
+		it.botHasPermissions(Permission.ViewChannel, Permission.SendMessages)
+	}.asChannelOfOrNull<GuildMessageChannel>()
+	return channelWithPerms
```

Comment:
This one can definitely be an `=` function
```suggestion
suspend inline fun getFirstUsableChannel(inputGuild: GuildBehavior): GuildMessageChannel? = inputGuild.channels.first {
		it.botHasPermissions(Permission.ViewChannel, Permission.SendMessages)
	}.asChannelOfOrNull<GuildMessageChannel>()
```